app/Http/Controllers/CustomFieldsetsController.php

```diff
@@ -93,16 +93,27 @@ public function store(Request $request)
     {
         $this->authorize('create', CustomField::class);
 
-        $cfset = new CustomFieldset([
+        $fieldset = new CustomFieldset([
                 'name' => e($request->get('name')),
                 'user_id' => Auth::user()->id,
         ]);
 
-        $validator = Validator::make($request->all(), $cfset->rules);
+        $validator = Validator::make($request->all(), $fieldset->rules);
+
         if ($validator->passes()) {
-            $cfset->save();
+            $fieldset->save();
+
+            // Sync fieldset with auto_add_to_fieldsets
+            $fields = CustomField::select('id')->where('auto_add_to_fieldsets', '=', '1')->get();
+            if ($fields->count() > 0) {
+                foreach ($fields as $field) {
+                    $field_ids[] = $field->id;
+                }
+
+                $fieldset->fields()->sync($field_ids);
+            }
```

Comment:
Side-note: since `$fields` is a collection you could skip the `foreach` and do
```suggestion
if ($fields->count() > 0) {
    $fieldset->fields()->sync($fields->pluck('id'));
}
```

Reply:
Are we sure what will work as expected - it feels to me more like it would *only* associate that one it's syncing to, unsyncing the others (which is why I set it up as an array).

Reply:
I think it would be ok. `$fields->pluck('id')` will return a collection of the ids, which can be cast to an array, in the same way `$field_ids[]` is formatted.

Definitely don't hold up the PR for this comment though 😄